frontend/src/lib/keybind.ts

```diff
@@ -0,0 +1,246 @@
+
+    const binding = this.bindings.find((binding) => binding.key === enum_key);
+    if (binding && binding.alt == alt && binding.shift == shift && binding.ctrl == ctrl) {
+      e.preventDefault();
+      binding.callback(e);
+    }
+  }
+}
+
+/*
+Search callbacks
+*/
+
+/**
+ * Used to indicate what direction {@link navigateResults} should go (up/down)
+ */
+type Direction = 1 | -1;
+
+/**
+ * Utilized to bring the next/previous result into focus
+ *
+ * @remarks
+ * Utilizes the element that is currently in focus and the given
+ * direction to determine where and how to traverse the array of
+ * results. Moves a single result per call.
+ *
+ * @param direction - How to maneuver through the result list (up/down), indicated by {@link Direction}
+ */
+const navigateResults = (direction: Direction) => {
+  const currentActive = document.activeElement;
+
+  const results = [...document.getElementsByClassName('result')] as HTMLElement[];
+  let currentResultIndex: number | null = null;
+  let newIndex = 0;
+
+  results.forEach((el, index) => {
+    if (el.contains(currentActive)) currentResultIndex = index;
+  });
+
+  if (currentResultIndex != null) {
+    newIndex = currentResultIndex + direction;
+    if (0 > newIndex || newIndex >= results.length) {
+      newIndex = currentResultIndex;
+    }
+  }
+
+  const nextResult = results[newIndex];
+
+  // Get the anchor element and focus it
+  (nextResult.getElementsByClassName('result-main-link')[0] as HTMLElement)?.focus();
+};
+
+/**
+ * Wrapper for {@link navigateResults} that will focus on the next result
+ */
+const focusNextResult = () => navigateResults(1);
+
+/**
+ * Wrapper for {@link navigateResults} that will focus on the previous result
+ */
+const focusPrevResult = () => navigateResults(-1);
+
+/**
+ * Focus the first result (if any)
+ */
+const focusMainResult = () => {
+  const results = document.getElementsByClassName('result-main-link');
+  if (results.length > 0) {
+    (results[0] as HTMLElement).focus();
+  }
+};
+
+/**
+ * Focus the search bar
+ */
+const focusSearchBar = () => {
+  (document.getElementById('searchbar') as HTMLInputElement)?.select();
+};
+
+/**
+ * Scroll to the top of the window and reset focus
+ */
+const scrollToTop = () => {
+  window.scrollTo({ top: 0, behavior: 'smooth' });
+  if (document.activeElement instanceof HTMLElement) {
+    document.activeElement.blur(); // reset focus
+  }
+};
+
+/**
+ * Redirect to the currently focused result
+ *
+ * @param e - The triggering {@link KeyboardEvent}
+ */
+const openResult = (e: KeyboardEvent) => {
+  if ((e.target as HTMLElement).className.includes('result-main-link')) {
+    const resultLink = e.target as HTMLAnchorElement;
+    window.location.href = resultLink.href;
+  } else {
+    const results = document.getElementsByClassName('result-main-link');
+    if (results.length > 0) {
+      window.location.href = (results[0] as HTMLAnchorElement).href;
+    }
+  }
+};
```

Comment:
could be written as
```ts
const openResult = (_e: KeyboardEvent, refs: Refs) => {
  const focusedResult: Result | undefined = refs.results.find((result) => result.hasFocus());
  if (focusedResult) {
    focusedResult.open();
  }
};
```
with related `.open()` and `.hasFocus()` functions in `Result.svelte`